scripting/chat-processor.sp

```diff
@@ -279,14 +294,16 @@ public void Frame_OnChatMessage_SayText2(any data)
 
 	if (iResults == Plugin_Changed)
 	{
-		for (int i = 0; i < GetArraySize(hRecipients); i++)
-		{
-			int client = GetArrayCell(hRecipients, i);
-
-			if (IsClientInGame(client))
-			{
-				CSayText2(client, sBuffer, iSender, bChat);
-			}
+		if(!bProto) {
+			Handle hMsg = StartMessage("SayText2", iRecipients, iNumRecipients, USERMSG_RELIABLE|USERMSG_BLOCKHOOKS);
+			
+			BfWriteByte(hMsg, iSender);
+			BfWriteByte(hMsg, bChat);
+			BfWriteString(hMsg, sBuffer);
+			EndMessage();
+		
+		} else {
+			PbSetString(bfMsg, "msg_name", sBuffer);
```

Comment:
Did you try copying the value of "textallchat" from the original protobuf message when creating your own here?

Might have to do some observations on the value if some consumer changes sFlag in the forward, so we adjust the value correctly.